engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateBy.java

```diff
@@ -1198,7 +1199,7 @@ public static Table updateBy(@NotNull final QueryTable source,
 
         final MutableObject<String> timestampColumnName = new MutableObject<>(null);
         // create an initial set of all source columns
-        final Set<String> preservedColumnSet = new LinkedHashSet<>(source.getColumnSourceMap().keySet());
+        final LinkedHashSet<String> preservedColumnSet = new LinkedHashSet<>(source.getDefinition().getColumnNames());
```

Comment:
Which is the right standard to follow here? We have at least one instance of "the old way", and a mix of list vs. set.
[](url)
![Screenshot 2023-11-13 at 7 35 58 PM](https://github.com/deephaven/deephaven-core/assets/5342661/1d2a1cf6-b2e7-447e-869c-17afb56c101b)

Reply:
The FormulaColumnPython example isn't the same interface. I'm happy to standardize on `getColumnNameSet` in these cases.

I did find another class of maps that could be optimized to sets as well.